### DISCUSSION TITLE
How can I force to the SWR don't hit the server if has already cached the key? (with a 15 minutes interval)

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [vercel/swr#570] [General]
Hi guys, 
I'm testing the swr and I want that the swr get data from the server only once, at the first page view, and then, only reload the data/cache after 15 minutes...

I noriced by the web dev console that the swr hit the server all the times when I view a page, even when the response is cached.

Can anyone helpe me?

Thanks so much!

**Comment by huozhi** [comment]
you can use the cache api to detect if the corresponding data is in the cache already

```js
import {cache} from 'swr'

useSWR(key, /*. .. */)

cache.has(key) // return `true` or `false`
```

**Reply by sergiodxa** [reply]
This will not prevent SWR to revalidate in focus

**Reply by huozhi** [reply]
yeah thanks @sergiodxa. sorry for the short snippet. I just listed out the API usage for user pick. the `cache.has` call can be anywhere we want. for fetching surly it could be put inside fetcher 👍

and if any other scenario requires to check if cache is fulfilled, `cache.has` is still able to leverage there : )

**Comment by sergiodxa** [comment]
I would move this to the fetcher, inside y check if data is already in cache and return it, also keep track somehow (another cache key you manually define maybe) when it was updated so you can check if it was longer than 15min to do the actual fetch or not to return cached value.

**Reply by pke** [reply]
I was stumbling about this myself again today. I would have thought that stable keys do not call the fetcher once the cache has been filled.
Especially in interdependent swr hooks it would fire multiple requests.

If you handle it inside the fetcher then how do you know when you should actually refetch? by comparing the handed in cache parts?

**Comment by lukebarton** [comment]
You could also let the brwoser cache it by using a `cache-control` header with a `max-age` of 15 minutes.

**Reply by lukebarton** [reply]
(providing you control the server issuing the header, of course)

**Comment by wilsonneto-swe** [comment]
Great guys, thanks!
I'm testing the dedupe option too, I think this will do this.

I'm doing some POCs yet.
Thanks so much for all the help.
